### PR TITLE
pin pandas>=2; stop supporting python 3.7; fix failing doctest

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description
- *Category*: bugfix
- *JIRA issue*: [MIC-3948](https://jira.ihme.washington.edu/browse/MIC-3948)

Doctests started failing with Pandas 2.0 and we temporarily pinned it at <2.0.
This goes all in with Pandas v2.0+.
- Pins Pandas>=2.0
- Removes support of python 3.7 (Pandas 2.0 requires python 3.8+)
- Updates readme 
- Fixes failing doctest

### Testing
CI is passing again

### CODE REVIEW

**Review by albrja — APPROVED** [review]

**Review by rmudambi — COMMENTED** [review]
Can you see if we can support 3.11?

File: README.rst
Comment (ramittal) [inline README.rst:1]: I am guessing, we generate the package version from the git-tag, would suggest to change major version number to signal this is a major change and does not break existing vivarium users.

Comment (stevebachmeier) [reply README.rst:1]: Yes, this is going into develop - we'll release in a separate ticket (and I agree it should be a major version change)

**Review by hussain-jafari — APPROVED** [review]

File: README.rst
Comment (rmudambi) [reply README.rst:1]: We don't need to get into this here, but I don't agree that this should be a major version change.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**